### PR TITLE
Add support for the `fix` option

### PR DESCRIPTION
**Pull request by gucong3000**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Why would you use `fix` in a gulp script?

**Comment by gucong3000** [comment]
I what fix files in git index, in pre-commit hook

File: readme.md [hunk 1]
```diff
@@ -35,12 +35,36 @@ gulp.task('default', () =>
...
 
 Type: `boolean`<br>
 Default: `false`
 
-Report errors only.
+When `true`, this option will filter warning messages from ESLint results. This mimics the ESLint CLI [quiet option](http://eslint.org/docs/user-guide/command-line-interface#quiet).
+
+Type: `function (message, index, list) { return Boolean(); }`
+
+When provided a function, it will be used to filter ESLint result messages, removing any messages that do not return a `true` (or truthy) value.
```
Comment (sindresorhus) [inline readme.md]: This is outside the scope of this option. I realize it comes from `gulp-eslint`, but it's not something I want exposed, at least not overloaded in this option.

File: test.js [hunk 2]
```diff
@@ -28,3 +28,20 @@ test('default formatter', async t => {
...
+	stream.on('data', file => {
+		t.is(file.contents.toString(), 'alert();\n');
+		t.pass();
+	});
+	stream.on('error', () => ({}));
+	stream.write(new gutil.File({
+		path: __filename,
+		contents: Buffer.from('alert()')
+	}));
+	stream.end();
```
Comment (sindresorhus) [inline test.js]: Use `p-event`

File: test.js [hunk 3]
```diff
@@ -28,3 +28,20 @@ test('default formatter', async t => {
 	stream.end();
 	await finish;
 });
+
+test('opts.fix', t => {
```
Comment (sindresorhus) [inline test.js]: Title: `fix option`

File: test.js (hunk 2 above)
Comment (gucong3000) [inline test.js]: Done.

**Comment by sindresorhus** [comment]
Looks good. Thanks for working on this :)

**Comment by sindresorhus** [comment]
Should I do a new release or do you plan on doing any more PRs first?

**Comment by gucong3000** [comment]
> do you plan on doing any more PRs?

No.
